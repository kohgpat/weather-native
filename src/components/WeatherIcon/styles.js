import styled from "styled-components/native";

const ENDPOINT = "https://openweathermap.org/img/w/";

export const Icon = styled.Image.attrs({
  source: props => {
    {
      uri: `${ENDPOINT}${props.icon}.png`;
    }
  }
})`
  width: 50px;
  height: 50px;
`;
