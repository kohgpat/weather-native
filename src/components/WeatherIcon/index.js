import React from "react";
import PropTypes from "prop-types";
import * as styles from "./styles";

const WeatherIcon = ({ weather }) => {
  if (!weather) {
    return null;
  }

  return <styles.Icon icon={weather.icon} />;
};

WeatherIcon.propTypes = {
  weather: PropTypes.object.isRequired
};

export default WeatherIcon;
