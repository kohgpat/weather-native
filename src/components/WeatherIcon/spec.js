/*global it, expect*/
import React from "react";
import WeatherIcon from "./index";
import { shallow } from "enzyme";

const weather = {
  icon: "10d"
};

it("renders", () => {
  const weatherIcon = shallow(<WeatherIcon weather={weather} />);
  expect(weatherIcon).toMatchSnapshot();
});
