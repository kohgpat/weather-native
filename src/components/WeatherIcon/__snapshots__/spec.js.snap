// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<styled.div
  icon="10d"
/>
`;
